sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -954,11 +954,18 @@ struct build_source_bundle_props;
 struct include_files
     : detail::run_time_property_key<include_files,
                                     detail::PropKind::IncludeFiles> {
-  include_files();
+  include_files() {}
   include_files(const std::string &name, const std::string &content) {
     record.emplace_back(std::make_pair(name, content));
   }
   void add(const std::string &name, const std::string &content) {
+    if (std::find_if(record.begin(), record.end(), [&name](auto &p) {
+          return p.first == name;
+        }) != record.end()) {
+      throw sycl::exception(make_error_code(errc::invalid),
+                            "Include file '" + name +
+                                "' is already registered");
+    }
```

Comment:
Does the specification promise this exception to be thrown?

It seems like potential overhead to go through the existing list over and over again.

Reply:
Could we potentially use a different data structure, e.g., map, to make the lookup more efficient and avoid the quadratic complexity?